src/Pauser.sol

```diff
@@ -0,0 +1,320 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.21;
+
+import "@openzeppelin-upgradeable/contracts/proxy/utils/Initializable.sol";
+import "@openzeppelin-upgradeable/contracts/proxy/utils/UUPSUpgradeable.sol";
+import "@openzeppelin-upgradeable/contracts/access/OwnableUpgradeable.sol";
+
+import "./interfaces/IPausable.sol";
+import "./RoleRegistry.sol";
+
+contract Pauser is Initializable, UUPSUpgradeable, OwnableUpgradeable {
+    // ========================================= STATE =========================================
+
+    /**
+     * @notice List of contracts that can be paused and unpaused using:
+     * - `pauseAll`
+     * - `unpauseAll`
+     */
+    IPausable[] public pausables;
+
+    /**
+     * @notice Used to pause calls to `deposit` and `depositWithPermit`.
+     */
+    bool public isPaused;
+
+    /**
+     * @notice Minimum delay required to execute an unpause action.
+     */
+    uint256 public minUnpauseDelay;
+
+    /**
+     * @notice Mapping of unpause operations and the time they become executable.
+     */
+    mapping(bytes32 id => uint256) public unpauseExecutionTime;
+
+    /**
+     * @notice Timestamp indicating an operation has been executed.
+     */
+    uint256 public constant _DONE_TIMESTAMP = uint256(1);
+
+    /**
+     * @notice Global Role registry contract.
+     */
+    RoleRegistry public roleRegistry;
+
+    //============================== ROLES ================================
+
+    /**
+     * @notice Contract specific roles.
+     */
+    bytes32 public constant PAUSER_ADMIN = keccak256("PAUSER_ADMIN");
+    /**
+     * @notice Protocol specific roles.
+     */
+    bytes32 public constant PROTOCOL_PAUSER = keccak256("PROTOCOL_PAUSER");
```

Comment:
not a problem in itself. Would become a tradeoff if this contract started nearing the size limit. Other potential footgun would be a situation where we changed the role on one contract but not the other. Unlikely but something we need to be aware of if you have it in 2 separate places and something we could possibly screw up in a future refactor. Not saying what you did is wrong, just that we are still feeling out this setup since we haven't used it before and we should think through situations like this

Reply:
Fetching directly from the role registry now.